compiler/ast/report_enums.nim

```diff
@@ -188,19 +188,24 @@ type
     rlexMalformedUnderscores
     rlexMalformedTrailingUnderscre
     rlexInvalidToken
+    rlexInvalidTokenSpaceBetweenNumAndIdent
     rlexNoTabs
 
     # numbers
-    rlexInvalidIntegerPrefix
+    rlexInvalidIntegerLiteralOctalPrefix
     rlexInvalidIntegerSuffix
     rlexNumberNotInRange
     rlexExpectedHex
     rlexInvalidIntegerLiteral
 
     # char
     rlexInvalidCharLiteral
+    rlexInvalidCharLiteralConstant
+    rlexInvalidCharLiteralPlatformNewline
+    rlexInvalidCharLiteralUnicodeCodepoint
     rlexMissingClosingApostrophe
-    rlexInvalidUnicodeCodepoint
+    rlexInvalidUnicodeCodepointEmpty
+    rlexInvalidUnicodeCodepointGreaterThan0x10FFFF
```

Comment:
```suggestion
    rlexInvalidUnicodeNotACodepoint
```
I think the problem is more clear this way.

Reply:
I'll revise this separately, I had something like that before and it felt vague... the current one is possibly too in the weeds. Either way, not a blocker as you indicated and eaisly fixiable.